### ISSUE TITLE
/pushgateway not exists in docker image

### ISSUE THREAD

**Issue by heww** [prometheus/pushgateway#106]
/pushgateway not exists in docker image which cause `prom/pushgateway` image run failed using unified container of mesos

```bash
I0308 17:45:46.894984  4941 exec.cpp:162] Version: 1.1.0
I0308 17:45:46.897234  4949 exec.cpp:237] Executor registered on agent f9dbe94e-845c-46aa-a97e-d4aee1fa9eb7-S0
Failed to chdir into current working directory '/pushgateway': No such file or directory
```

**Comment by beorn7** [comment]
@discordianfish Is this something you could help with? I'm still pretty Docker illiterate...

**Comment by grobie** [comment]
Might be the same problem with CGO again @sdurrheimer?

**Comment by discordianfish** [comment]
@heww This appears to be a problem on the mesos side: The universal containerizer isn't creating the working directory. The Docker image can be run without problems: `docker run prom/pushgateway`. Cloudfoundary has a similar issue: https://github.com/cloudfoundry-attic/garden-linux/issues/29. I would suggest you open an issue with mesos.

@beorn7 Maybe it's time to learn some Docker. ;)

As a workaround we could also create the directory in the Dockerfile.

**Comment by sdurrheimer** [comment]
Technically, docker should create the working directory if it doesn't exists before chdir into it.
I'm don't know much about Mesos, but if I understand this correctly, it doesn't really run a docker-engine, but is able to use the docker image format to run containers ? 

So I guess this is a Mesos issue, but if we want to, we could manually create the directory in the Dockerfile to fix this edge case.

**Comment by discordianfish** [comment]
@sdurrheimer correct, that universal executor is their own implementation. Dunno if the behavior of creating the directory is in the spec though.
